core/logic.py

```diff
@@ -0,0 +1,74 @@
+from context import models
```

Comment:
There is lack of OOP in your project. You need to define functions in classes so that these may be available to you later through inheritance.